util/ranger/new_refiner.go

```diff
@@ -21,28 +21,34 @@ import (
 	"github.com/pingcap/tidb/util/types"
 )
 
-func buildIndexRange(sc *variable.StatementContext, cols []*expression.Column, lengths []int, inAndEqCount int,
+func buildIndexRange(sc *variable.StatementContext, cols []*expression.Column, lengths []int,
 	accessCondition []expression.Expression) ([]*types.IndexRange, error) {
 	rb := builder{sc: sc}
-	var ranges []*types.IndexRange
-	for i := 0; i < inAndEqCount; i++ {
+	var (
+		ranges  []*types.IndexRange
+		eqCount int
+	)
+	for eqCount = 0; eqCount < len(accessCondition) && eqCount < len(cols); eqCount++ {
+		if sf, ok := accessCondition[eqCount].(*expression.ScalarFunction); !ok || sf.FuncName.L != ast.EQ {
```

Comment:
The original is `inAndEqCount`, now only `Eq` is counted.

Reply:
`in` has be rewritten to `or`. Currently there won't be any `in` function any more.